rusk-wallet/src/wallet.rs

```diff
@@ -588,43 +567,61 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
         addr: &Address,
         gas: Gas,
     ) -> Result<Transaction, Error> {
-        let wallet = self.connected_wallet().await?;
+        let state = self.state()?;
         // make sure we own the staking address
         if !addr.is_owned() {
             return Err(Error::Unauthorized);
         }
 
         let mut rng = StdRng::from_entropy();
-        let index = addr.index()? as u64;
+        let index = addr.index()?;
+
+        let sender_sk = derive_phoenix_sk(&self.seed, index);
+        let stake_sk = derive_bls_sk(&self.seed, index);
 
-        let tx = wallet.withdraw(
+        let inputs = state
+            .inputs(index, gas.limit * gas.price)?
+            .into_iter()
+            .map(|(note, opening)| {
+                let nullifier = note.gen_nullifier(&sender_sk);
+
+                (note, opening, nullifier)
+            })
+            .collect();
+
+        let root = state.fetch_anchor()?;
+
+        let reward_amount =
+            state.fetch_stake(addr.pk())?.map(|s| s.reward).unwrap_or(0);
+
+        let withdraw = phoenix_stake_reward::<_, Prover>(
             &mut rng,
-            index,
-            index,
-            addr.psk(),
+            &sender_sk,
+            &stake_sk,
+            inputs,
+            root,
+            reward_amount,
             gas.limit,
             gas.price,
         )?;
-        Ok(tx)
+
+        state.propagate(withdraw)
     }
 
     /// Returns bls key pair for provisioner nodes
     pub fn provisioner_keys(
         &self,
         addr: &Address,
-    ) -> Result<(PublicKey, SecretKey), Error> {
+    ) -> Result<(PhoenixPublicKey, PhoenixSecretKey), Error> {
```

Comment:
provisioner keys are bls keys

Reply:
Changed to `BlsPublicKey` and `BlsSecretKey`